keanu-project/src/test/java/io/improbable/keanu/tensor/dbl/DoubleTensorTest.java

```diff
@@ -313,10 +496,74 @@ private void assertAllValuesAre(DoubleTensor tensor, double v) {
         }
     }
 
+    @Test
+    public void canTranspose() {
+        DoubleTensor a = DoubleTensor.create(new double[]{1, 2, 3, 4}, new long[]{2, 2});
+        DoubleTensor actual = a.transpose();
+        DoubleTensor expected = DoubleTensor.create(new double[]{1, 3, 2, 4}, 2, 2);
+
+        assertEquals(expected, actual);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
```

Comment:
in those tests that have `expected = ...`: can you assert something about the error message too? (using the `ExpectedException` object which you already have in this class)